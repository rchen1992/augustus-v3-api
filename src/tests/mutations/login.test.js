const { testQuery } = require('@tests/utils');

const userId = 10;
const userName = 'newboi';

testQuery({
    id: 'Log in an existing user',
    query: `
        mutation login($input: LoginInput!) {
            login(input: $input) {
                id
                userName
                email
                createdAt
            }
        }
    `,
    variables: {
        input: {
            userName,
            password: 'password1',
        },
    },
    context: () => {
        return {
            services: {
                auth: {
                    login() {
                        return {
                            user_id: userId,
                            user_name: userName,
                            email: 'newboi@gmail.com',
                            created_at: '2018-12-11 09:39:31',
                            updated_at: '2018-10-09 01:00:11',
                        };
                    },
                },
            },
        };
    },
    expected: undefined,
    additionalAssertions(result, context) {
        expect(context.req.session.userId).toBe(userId);
    },
});
