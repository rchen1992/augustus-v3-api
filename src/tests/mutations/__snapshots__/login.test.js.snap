// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query: Log in an existing user 1`] = `
Object {
  "data": Object {
    "login": Object {
      "createdAt": "2018-12-11 09:39:31",
      "email": "newboi@gmail.com",
      "id": "10",
      "userName": "newboi",
    },
  },
}
`;
